modules/mod_logging/models/m_log_ui.erl

```diff
@@ -0,0 +1,215 @@
+%% @author Marc Worrell
+%% @copyright 2019 Marc Worrell
+%% @doc Model for ui log messages.
+
+%% Copyright 2019 Marc Worrell
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(m_log_ui).
+-author("Marc Worrell").
+
+-behaviour(gen_model).
+
+%% interface functions
+-export([
+    m_find_value/3,
+    m_to_list/2,
+    m_value/2,
+    insert_event/2,
+    insert/2,
+    get/2,
+    search_query/2,
+    periodic_cleanup/1,
+    install/1
+]).
+
+
```

Comment:
Sometimes there are double newlines and sometimes single ones, is there a logic behind it?

Reply:
No, sometimes I just want to have a bit more "visual space" between functional different parts. Helps with reading and grouping.